pkg/api/scylla/v1alpha1/types_scylladbmanagertask.go

```diff
@@ -0,0 +1,179 @@
+	// +optional
+	SnapshotParallel []string `json:"snapshotParallel,omitempty"`
+
+	// uploadParallel specifies a list of upload parallelism limits in the following format: `[<dc>:]<limit>`.
+	// `<dc>:` is optional and allows for specifying different limits in selected datacenters. If `<dc>:` is not set, the limit is global.
+	// For instance, `[]string{"dc1:2", "5"}` corresponds to two parallel nodes in `dc1` datacenter and five parallel nodes in the other datacenters.
+	// +optional
+	UploadParallel []string `json:"uploadParallel,omitempty"`
+}
+
+type ScyllaDBManagerRepairTaskOptions struct {
+	// schedule specifies the schedule on which the repair task is run.
+	ScyllaDBManagerTaskSchedule `json:",inline"`
+
+	// dc specifies a list of datacenter `glob` patterns, e.g. `dc1`, `!otherdc*`, determining the datacenters to include or exclude from repair.
+	// +optional
+	DC []string `json:"dc,omitempty"`
+
+	// keyspace specifies a list of `glob` patterns used to include or exclude tables from repair.
+	// The patterns match keyspaces and tables. Keyspace names are separated from table names with a dot e.g. `!keyspace.table_prefix_*`.
+	// +optional
+	Keyspace []string `json:"keyspace,omitempty"`
+
+	// failFast indicates that a repair should be stopped on first encountered error.
+	// +optional
+	FailFast *bool `json:"failFast,omitempty"`
+
+	// host specifies the IPv4 or IPv6 address of a node to repair.
+	// Specifying this field limits repair to token ranges replicated by a given node.
+	// When used in conjunction with `dc`, the node must belong to the specified datacenters.
+	// If not set, all hosts are repaired.
+	// +optional
+	Host *string `json:"host,omitempty"`
+
+	// intensity specifies the number of token ranges to repair in a single ScyllaDB node at the same time.
+	// Changing the intensity impacts the repair granularity in case it is resumed. The higher the value, the more work on resumption.
+	// When set to zero, the number of token ranges is adjusted to the maximum supported number.
+	// When set to a value greater than the maximum supported by the node, intensity is capped at the maximum supported value.
+	// Refer to repair documentation for details.
+	// +optional
+	Intensity *int64 `json:"intensity,omitempty"`
+
+	// parallel specifies the maximum number of ScyllaDB repair jobs that can run at the same time (on different token ranges and replicas).
+	// Each node can take part in at most one repair at any given moment. By default, the maximum possible parallelism is used.
+	// The maximal effective parallelism depends on keyspace replication strategy and cluster topology.
+	// When set to a value greater than the maximum supported by the node, parallel is capped at the maximum supported value.
+	// Refer to repair documentation for details.
+	// +optional
+	Parallel *int64 `json:"parallel,omitempty"`
```

Comment:
Shouldn't we verify this is not negative?

Reply:
Well, SM does not check for negative parallel. Moreover, for negative parallel, it behaves as advertised - the repair gets stuck because no repair jobs can run... We should validate it on SM side in order to protect user from themselves.